server/src/main/java/org/apache/cassandra/sidecar/db/SidecarLeaseDatabaseAccessor.java

```diff
@@ -76,20 +78,22 @@ public LeaseClaimResult extendLease(String currentOwner)
      */
     public static class LeaseClaimResult
     {
-        public final boolean leaseAcquired;
         public final String currentOwner;
 
-        LeaseClaimResult(boolean leaseAcquired, String currentOwner)
+        LeaseClaimResult(String currentOwner)
         {
-            this.leaseAcquired = leaseAcquired;
             this.currentOwner = currentOwner;
         }
 
         static LeaseClaimResult from(ResultSet resultSet, String newOwner)
         {
             return resultSet.wasApplied()
-                   ? new LeaseClaimResult(true, newOwner)
-                   : new LeaseClaimResult(false, resultSet.one().getString("owner"));
+                   ? new LeaseClaimResult(newOwner)
+                   // In some rare cases, the resultSet will not contain the owner information
+                   // even though the resultSet was not applied. This will translate into an
+                   // IllegalArgumentException being thrown when trying to retrieve the non-existing
+                   // owner string. This exception is left to be handled by the caller method
+                   : new LeaseClaimResult(resultSet.one().getString("owner"));
```

Comment:
It is unclear to me what is the proper handling at the caller. Maybe handle exception here and set a new state, say "unknown". Caller can react on the "unknown" state, for example, query retry.